catroid/src/main/java/org/catrobat/catroid/web/ServerCalls.java

```diff
@@ -817,6 +817,16 @@ public boolean login(String username, String password, String token, Context con
 			String serverAnswer = jsonObject.optString(JSON_ANSWER);
 			SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
 
+			if (BuildConfig.CREATE_AT_SCHOOL) {
+				boolean createAtSchoolUser = jsonObject.optBoolean(Constants.CREATE_AT_SCHOOL_USER);
+				sharedPreferences.edit().putBoolean(Constants.CREATE_AT_SCHOOL_USER, createAtSchoolUser).apply();
+				if (!createAtSchoolUser) {
+					sharedPreferences.edit().putString(Constants.TOKEN, Constants.NO_TOKEN).apply();
```

Comment:
@sjaindl please check that, after this we can merge, if your next PR is the bugfix for invalid entries ;)